### PR TITLE
perf: Remove recursive calling code

### PR DESCRIPTION
**Pull request by zgfzgf**
Remove recursive calling code

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1645)

### CODE REVIEW

**Comment by iphydf** [comment]
Hi @zgfzgf, this looks decent. Can you fix the formatting according to restyled (and astyle in Travis CI)? The "asan" test failure isn't related to your change, so you can ignore it (I need to figure out why this keeps happening, but you can ignore it for now).

We also have a check to limit the number of recursive calls in toxcore, which will probably fail now because you've eliminated one instance of recursion (ideally we have no recursion at all anywhere). Look at other/analysis/check_recursion for the list of recursive functions.

**Comment by zgfzgf** [comment]
Is this okay? @iphydf

**Comment by zgfzgf** [comment]
I do not know what this means, what is the specific format?
? Validator: TITLE
? (Semantic release conventions must be followed. OR Dependabot PRs are exempt from semantic release conventions.)
Input : Restyle perf: Remove recursive calling code
Settings : {"or":[{"must_include":{"regex":"^(feat|docs|chore|cleanup|fix|refactor|test|style|perf)(\\(\\w+\\))?:\\ .+$","message":"Semantic release conventions must be followed."}},{"must_include":{"regex":"^Bump [^ ]* from [^ ]* to [^ ]*$","message":"Dependabot PRs are exempt from semantic release conventions."}}]}

**Comment by zgfzgf** [comment]
Failing after 1m — Build Failed is + Blank line

File: toxcore/DHT.c [hunk 1]
```diff
@@ -753,23 +753,29 @@ static int client_or_ip_port_in_list(const Logger *log, const Mono_Time *mono_ti
...
                  const uint8_t *cmp_pk)
 {
+    bool ret = false;
+    uint8_t pk_new[CRYPTO_PUBLIC_KEY_SIZE];
+    memcpy(pk_new, pk, CRYPTO_PUBLIC_KEY_SIZE);
+    IP_Port ip_port_new = ip_port;
+
+    uint8_t pk_bak[CRYPTO_PUBLIC_KEY_SIZE];
+    IP_Port ip_port_bak;
+
```
Comment (iphydf) [inline toxcore/DHT.c:763]: This line contains trailing whitespace. Please remove it.

**Comment by zgfzgf** [comment]
"add_to_list -> add_to_list" in  other/analysis/check_recursion 
@iphydf Do you want to delete it

**Comment by zgfzgf** [comment]
@iphydf, I want to correct the recursion of another function add_to_closest. Is it possible to close this request?

**Comment by iphydf** [comment]
Looks ok, but this code needs a unit test. I'll write one at some point. If you feel like it, please write one in DHT_test.cc (doesn't exist yet, copy from util_test.cc).

**Comment by iphydf** [comment]
I've added one here: https://github.com/TokTok/c-toxcore/pull/1713

**Comment by iphydf** [comment]
@zgfzgf it's failing the test, which means this implementation does not behave the same way as the old one. Is that intentional?

**Comment by iphydf** [comment]
Closing for inactivity.

**Pull request closed without merging**